### PR TITLE
Feat/on chain json validation rebase/111

### LINKED ISSUE #111: On-chain Schema JSON Validation Prior to Registration
As a user, I should NOT be able to register an malformed schema so that all schemas registered on-chain are valid JSON

## Acceptance Criteria
- [x] Schema with invalid JSON should be rejected from being registered
- [x] Schema with valid JSON should be allowed to be registered

## Notes
- [serde](https://serde.rs/) parsing

This replaces https://github.com/LibertyDSNP/mrc/issues/29 for now.

### PR DESCRIPTION
**Pull request by mehtaishita**
# Goal
The goal of this PR is to add Json model validation on chain

Closes #111 

# Discussion
Null or empty Json schemas will be invalidated and not registered (even though serde-json does not consider them invalid)

# Checklist
- [ ] Chain spec updated
- [ ] Updated the js/rpc code if a custom RPC added/changed 
- [ ] Design doc(s) updated
- [x] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

**Comment by mehtaishita** [comment]
Comments from previous PR:

[x] Add negative json validation test
[x] Add utf 8 encoding test
[x] Clean up files changed
[x] Move serde to schema pallet since it needs to be no-std

Discussion:
Do I need to provide feedback to the user trying to register their schema as to why their schema failed to register? 
The only error code that exists on the schema pallet currently is 'invalid schema'

File: pallets/schemas/src/mock.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ impl WeightToFeePolynomial for WeightToFee {
 impl pallet_schemas::Config for Test {
 	type Event = Event;
 	type WeightInfo = ();
-	type MinSchemaModelSizeBytes = ConstU32<5>;
+	type MinSchemaModelSizeBytes = ConstU32<15>;
```
Comment (dopatraman) [inline pallets/schemas/src/mock.rs:56]: Just curious, what does `15` represent here?

**Review by dopatraman — APPROVED** [review]

File: pallets/schemas/src/mock.rs (hunk 1 above)
Comment (mehtaishita) [reply pallets/schemas/src/mock.rs:56]: the number of bytes of string representation of the schema model to be registered

**Review by enddynayn — APPROVED** [review]
Nice! 👍

File: pallets/schemas/src/lib.rs [hunk 2]
```diff
@@ -66,6 +66,7 @@ pub use pallet::*;
 pub mod weights;
 pub use types::*;
 pub use weights::*;
+mod serde;
```
Comment (enddynayn) [inline pallets/schemas/src/lib.rs:69]: do you need to do:
```
pub use serde:: validate_json_model
```

**Pull request merged**